src/orion/core/worker/experiment_config.py

```diff
@@ -0,0 +1,135 @@
+""" Immutable dataclass containing the experiment configuration. """
+from __future__ import annotations
+
+import typing
+from typing import Any, TypedDict
+
+if typing.TYPE_CHECKING:
+    from orion.core.evc.adapters import CompositeAdapter
+
+
+class EmptyDict(TypedDict):
+    """Empty dictionary."""
+
+
+class DatabaseConfig(TypedDict):
+    """Database configuration."""
+
+    type: str
+
+
+class StorageConfig(TypedDict):
+    """Storage configuration."""
+
+    type: str
+    database: DatabaseConfig
+
```

Comment:
We are these used?

Reply:
Nowhere atm, since (afaik, but correct me if I'm wrong) the storage isn't part of the Experiment configuration dictionary.
However, if it is, or if it becomes part of it, we could then add a `storage: StorageConfig` annotation onto the `ExperimentConfig` typeddict.

Reply:
I think we should add them when we need them.

Reply:
Removed in 8f48a0e6ef9bb451fee59dbdf7e0f07d1be2f66e

Reply:
ok thanks!